### PR TITLE
[BugFix] Disable in-tablet parallel, add counter and get index timeout for pointer query using pk index

### LINKED ISSUE #25705: Point query using pk index should disable in tablet parallelism
### Real behavior (Required)
Point query using pk index should disable in tablet parallelism, or the result may be duplicated.
```
mysql> select * from pkdemo [_USE_PK_INDEX_] where pk = 92613258;
+----------+----------+
| pk       | sort_col |
+----------+----------+
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
| 92613258 |     4046 |
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by decster**
Fixes #25705

This PR disables in-tablet parallel, adds counter and get index timeout for pointer query.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/storage/rowset/segment_iterator.cpp [hunk 1]
```diff
@@ -392,9 +392,13 @@ Status SegmentIterator::_init() {
     RETURN_IF_ERROR(_get_row_ranges_by_keys());
     RETURN_IF_ERROR(_get_row_ranges_by_rowid_range());
     RETURN_IF_ERROR(_apply_del_vector());
-    RETURN_IF_ERROR(_apply_bitmap_index());
-    RETURN_IF_ERROR(_get_row_ranges_by_zone_map());
-    RETURN_IF_ERROR(_get_row_ranges_by_bloom_filter());
+    // skip some index filters if the scan range is already empty or single row.
+    // single row case happens when query is a pointer query and we already use pk index to get a single row range
+    if (!_scan_range.is_single_row_or_empty()) {
```
Comment (chaoyli) [inline be/src/storage/rowset/segment_iterator.cpp:397]: If the query is a point query, the `bitmap index` can helps for not primary key table.

**Review by luohaha — APPROVED** [review]

File: be/src/storage/rowset/segment_iterator.cpp (hunk 1 above)
Comment (decster) [reply be/src/storage/rowset/segment_iterator.cpp:397]: in that case, `!_scan_range.is_single_row_or_empty()` will be true

Comment (decster) [reply be/src/storage/rowset/segment_iterator.cpp:397]: not primary key table, `!_scan_range.is_single_row_or_empty()` will be true

**Review by chaoyli — APPROVED** [review]

File: be/src/storage/rowset/segment_iterator.cpp (hunk 1 above)
Comment (decster) [reply be/src/storage/rowset/segment_iterator.cpp]: removed

**Review by Astralidea — APPROVED** [review]

**Comment by decster** [comment]
@mergify backport branch-3.1

**Pull request merged**